### PR TITLE
feat(gh-32): helmRelease resource

### PR DESCRIPTION
**Pull request by MadhuvanthG**
- Initial commit for  depscloud/deps.cloud#32
- Got the services deployed using this `HelmResource` locally in Minikube

### CODE REVIEW

File: helmrelease/depscloud-system.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: depscloud-system
+spec:
+  helmVersion: v3
+  releaseName: depscloud-system
+  chart:
+    git: https://github.com/depscloud/deploy.git
+    ref: main
+    path: charts/depscloud
+
```
Comment (MadhuvanthG) [inline helmrelease/depscloud-system.yaml:12]: Does our umbrella chart accept `values` for image of different services (tracker, indexer etc..). I'm wondering, when we run integrations tests on a PR in tracker for example, how to make the tracker service deployed by the chart use the ephemeral image.

Comment (mjpitz) [reply helmrelease/depscloud-system.yaml:12]: Yup the `values` for the umbrella chart are as such:

```
tracker:
  enabled: true
  image:
    repository: x.b.c/group/depscloud-tracker
    tag: abc1234
```

This same structure applies to all other depscloud services (`gateway`, `extractor`, `indexer`). This README provides some of that information without requiring duplication from each other chart.

https://github.com/depscloud/deploy/tree/main/charts/depscloud#parameters

File: helmrelease/depscloud-system.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+kind: HelmRelease
+metadata:
+  name: depscloud-system
+spec:
+  helmVersion: v3
+  releaseName: depscloud-system
+  chart:
+    git: https://github.com/depscloud/deploy.git
+    ref: main
+    path: charts/depscloud
```
Comment (mjpitz) [inline helmrelease/depscloud-system.yaml:11]: We should point this at the depscloud hosted helm repository instead. See here for how to reference the helm repo: https://deps.cloud/docs/deployment/helm/#using-helm-operator

File: helmrelease/depscloud-system.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: depscloud-system
+spec:
+  helmVersion: v3
+  releaseName: depscloud-system
```
Comment (mjpitz) [inline helmrelease/depscloud-system.yaml:7]: I'd just name this depsloud

File: helmrelease/depscloud-system.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: depscloud-system
```
Comment (mjpitz) [inline helmrelease/depscloud-system.yaml:4]: depscloud

File: helmrelease/crds.yaml [hunk 5]
```diff
@@ -0,0 +1,414 @@
+---
```
Comment (mjpitz) [inline helmrelease/crds.yaml:1]: Committing this is probably fine for now. I know that the operator is able to install this on your behalf so we don't have to manage our own copies.

**Comment by mjpitz** [comment]
Off to a great start.

File: helmrelease/crds.yaml (hunk 5 above)
Comment (MadhuvanthG) [reply helmrelease/crds.yaml:1]: Ah okay, I didn't know that. Yeah that's better, so we don't have the overhead of maintaining a copy.

File: helmrelease/depscloud-system.yaml (hunk 2 above)
Comment (MadhuvanthG) [reply helmrelease/depscloud-system.yaml:11]: Sure, sounds good. Btw, didn't know we had a documentation on that. Nice!

**Comment by MadhuvanthG** [comment]
@mjpitz, addressed the changes. Will remove the crd's file later, if that's okay.

**Pull request merged**